### ISSUE TITLE
The generated docs don't work

### ISSUE THREAD

**Issue by mcmire** [thoughtbot/shoulda-matchers#1139]
Due to changes in browsers, it seems like the generated docs at http://matchers.shoulda.io/docs don't seem to work. I don't think this should block the release for 4.0.0, but it's something that probably should be fixed so people have something nice to look at. I briefly looked into it just now and it seems like this isn't as easy to fix as I thought. YARD has made some updates to their [`default` template](https://github.com/lsegal/yard/tree/master/templates/default) that fix the JavaScript errors that are happening now, but they're incompatible with the changes I made in creating the current layout. It's been so long since I've touched that code, I completely forgot how YARD templates work. So I'll have to spend some time fixing this :(

**Comment by geoffharcourt** [comment]
@mcmire the current docs show some things in master from January, but that haven't been released yet (`allow_nil` on `validates_length_of` is an example). Could a minor release get cut before 4.0.0 so that the current docs would reflect what's already available?

**Comment by mcmire** [comment]
Yup! We do plan on releasing an RC soon. This PR, by the way, will be actually fixed by #1143, I just didn't link it. But that's a good point about removing the 4.0.0 docs. I've fixed this for now, although now that I look at it, it seems that the docs for 3.1.2 never got generated. I'll fix this shortly.
